### PR TITLE
Entrenching Tools can now be saved in loadouts and will not pick up snow piles while folded.

### LINKED ISSUE #7317: Shovels are not saved on loadout.
### Description

Shovels arent saved on loadout, causing construction pouches to not spawn with them on loadouts system

### Test Merges

#7164
#7257
#7194
#7099

### Reproduction Steps

Create a loadout with a construction pouch (starts with a shovel inside of it) and then load that loadout. There will be no shovel in the construction pouch

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by Arctrooper8**
## About The Pull Request

Fixes #7317 https://github.com/tgstation/TerraGov-Marine-Corps/issues/7317
This is accomplished by making shovels start folded, I made sure to test for any unintended side effects and found none but I did find a new bug: 
A folded shovel can pick up a snow pile, not the snow on the ground but the snow pile item itself. This bug has also been fixed with an if else check in the snow file.

## Why It's Good For The Game

Bugfixes are always good.

## Changelog
:cl:
fix: Entrenching tools now start folded and as such will now save in loadouts.
fix: Folded Entrenching tools can no longer pick up snow piles.
/:cl:

### CODE REVIEW

**Comment by Luxyboo** [comment]
make it so i dont have to shovel 5 snow tiles for a single cade

**Comment by Arctrooper8** [comment]
> make it so i dont have to shovel 5 snow tiles for a single cade

That'd have to be a different PR but not a bad idea, especially with snow being limited sometimes. Unlike the magic let me pull 100 dirt from one tile big red and LZ ground. Filling sandbags honestly makes them super situational.

File: code/game/objects/items/stacks/snow.dm [hunk 1]
```diff
@@ -18,35 +18,37 @@
...
+				transf_amt = amount
 
+			ET.dirt_amt = transf_amt
+			ET.dirt_type = DIRT_TYPE_SNOW
+			to_chat(user, span_notice("You take snow from [src]."))
+			ET.update_icon()
+			use(transf_amt)
+			return TRUE
+		else
+			to_chat(user, span_warning("You must unfold your shovel first!"))
```
Comment (TiviPlus) [inline code/game/objects/items/stacks/snow.dm:51]: use early returns

**Review by TiviPlus — CHANGES_REQUESTED** [review]
in future please split fixes so that its faster and easier to review

**Comment by Arctrooper8** [comment]
It now uses an early return. And in the future I shall atomize even further with fixes.

**Comment by Arctrooper8** [comment]
> make it so i dont have to shovel 5 snow tiles for a single cade

https://github.com/tgstation/TerraGov-Marine-Corps/pull/9719 Tada, thanks for the idea.

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**